esp-hal/src/uart.rs

```diff
@@ -451,18 +461,28 @@ pub enum ConfigError {
     ///    deviation to the desired value, and the driver cannot reach this
     ///    speed.
     UnsupportedBaudrate,
+
     /// The requested  timeout exceeds the maximum value (
     #[cfg_attr(esp32, doc = "127")]
     #[cfg_attr(not(esp32), doc = "1023")]
     /// ).
     UnsupportedTimeout,
+
     /// The requested RX FIFO threshold exceeds the maximum value (
     #[cfg_attr(esp32, doc = "127")]
     #[cfg_attr(any(esp32c6, esp32h2), doc = "255")]
     #[cfg_attr(any(esp32c2, esp32c3, esp32s2), doc = "511")]
     #[cfg_attr(esp32s3, doc = "1023")]
     /// ).
     UnsupportedRxFifoThreshold,
+
+    /// The requested TX FIFO threshold exceeds the maximum value (
+    #[cfg_attr(esp32, doc = "127")]
+    #[cfg_attr(any(esp32c6, esp32h2), doc = "255")]
+    #[cfg_attr(any(esp32c2, esp32c3, esp32s2), doc = "511")]
+    #[cfg_attr(esp32s3, doc = "1023")]
```

Comment:
Is this 1023 max only the case when UARTX uses _all_ the available UART FiFo memory? The default will still be 127 I think? Not sure how to approach documenting this.

Reply:
And I suppose this also applies to rx fifo thresholds, it's all dependent on the amount of FiFo memory used (on some chips)

Reply:
This is difficult because these numbers are what fit in the registers. I guess we can shrink everything to 128 bytes (and 127 as limit) for now.